src/pow.cpp

```diff
@@ -239,6 +239,10 @@ unsigned int GetNextWorkRequiredV4(const CBlockIndex* pindexLast, const Consensu
         {
             bnNew *= (100 + params.nLocalTargetAdjustment);
             bnNew /= 100;
+            if (i % 16 == 0 && bnNew > UintToArith256(params.powLimit)) {
```

Comment:
Should add some comments regarding the this logic and mod 16.  This is consensus code.